htdocs/takepos/invoice.php

```diff
@@ -1871,229 +1871,242 @@ function SetNote() {
-							}
-							$htmlforlines .= $line->qty;
-							$htmlforlines .= '&nbsp; ';
-							$htmlforlines .= '<span class="opacitylow" title="'.$langs->trans("Stock").' '.price($stock_real, 1, '', 1, 0).'">';
-							$htmlforlines .= '(';
-							if ($line->qty && $line->qty > $stock_real) {
-								$htmlforlines .= '<span style="color: var(--amountremaintopaycolor)">';
-							}
-							$htmlforlines .= img_picto('', 'stock', 'class="pictofixedwidth"').price($stock_real, 1, '', 1, 0);
-							if ($line->qty && $line->qty > $stock_real) {
-								$htmlforlines .= "</span>";
-							}
-							$htmlforlines .= ')';
-							$htmlforlines .= '</span>';
-						} else {
-							dol_print_error($db);
-						}
-					} else {
-						$htmlforlines .= $line->qty;
-					}
-				} else {
-					$htmlforlines .= $line->qty;
-				}
-
-				$htmlforlines .= '</td>';
-				if (getDolGlobalInt('TAKEPOS_SHOW_HT')) {
-					$htmlforlines .= '<td class="right classfortooltip" title="'.$moreinfo.'">';
-					$htmlforlines .= price($line->total_ht, 1, '', 1, -1, -1, $conf->currency);
-					if (isModEnabled('multicurrency') && !empty($_SESSION["takeposcustomercurrency"]) && $conf->currency != $_SESSION["takeposcustomercurrency"]) {
-						//Only show customer currency if multicurrency module is enabled, if currency selected and if this currency selected is not the same as main currency
-						include_once DOL_DOCUMENT_ROOT.'/multicurrency/class/multicurrency.class.php';
-						$multicurrency = new MultiCurrency($db);
-						$multicurrency->fetch(0, $_SESSION["takeposcustomercurrency"]);
-						$htmlforlines .= '<br><span id="linecolht-span-total" style="font-size:0.9em; font-style:italic;">('.price($line->total_ht * $multicurrency->rate->rate).' '.$_SESSION["takeposcustomercurrency"].')</span>';
-					}
-					$htmlforlines .= '</td>';
-				}
-				$htmlforlines .= '<td class="right classfortooltip" title="'.$moreinfo.'">';
-				$htmlforlines .= price($line->total_ttc, 1, '', 1, -1, -1, $conf->currency);
-				if (isModEnabled('multicurrency') && !empty($_SESSION["takeposcustomercurrency"]) && $conf->currency != $_SESSION["takeposcustomercurrency"]) {
-					//Only show customer currency if multicurrency module is enabled, if currency selected and if this currency selected is not the same as main currency
-					include_once DOL_DOCUMENT_ROOT.'/multicurrency/class/multicurrency.class.php';
-					$multicurrency = new MultiCurrency($db);
-					$multicurrency->fetch(0, $_SESSION["takeposcustomercurrency"]);
-					$htmlforlines .= '<br><span id="linecolht-span-total" style="font-size:0.9em; font-style:italic;">('.price($line->total_ttc * $multicurrency->rate->rate).' '.$_SESSION["takeposcustomercurrency"].')</span>';
-				}
-				$htmlforlines .= '</td>';
-			}
-			$htmlforlines .= '</tr>'."\n";
-			$htmlforlines .= empty($htmlsupplements[$line->id]) ? '' : $htmlsupplements[$line->id];
-
-			print $htmlforlines;
+            // complete line by hook
```

Comment:
You should install pre-commit  
See https://wiki.dolibarr.org/index.php?title=Environment_and_development_tools#Install_Pre-commit_githook
So such error will never appear in future.